pkg/hostagent/snats.go

```diff
@@ -1200,15 +1206,20 @@ func (agent *HostAgent) isPolicyNameSpaceMatches(policyName string, namespace st
 }
 
 func (agent *HostAgent) getSnatUuids(poduuid string) []string {
+	localInfo := &opflexSnatLocalInfo{}
+	plcyUuids := []string{}
 	agent.indexMutex.Lock()
+	defer agent.indexMutex.Unlock()
 	val, check := agent.opflexSnatLocalInfos[poduuid]
-	agent.indexMutex.Unlock()
 	if check {
-		return val.PlcyUuids
-
-	} else {
-		return []string{}
+		err := util.DeepCopyObj(val, localInfo)
+		if err != nil {
+			agent.log.Error(err.Error())
+			return plcyUuids
```

Comment:
Why are we returning the plcyUuids even if the deep copy fails? We should be returning nil. In general it is best practice to have the return value([]string) and the error in the func definition, so we can return both. For this case, when there is an err we return nil, err and when the func runs sucessfully we return plcyUuids, nil

Reply:
addressed